src/Utils.vala

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2022 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+namespace Utils {
+    public static bool is_running_in_demo_mode () {
+        var proc_cmdline = File.new_for_path ("/proc/cmdline");
```

Comment:
Since this is only used in `indicator.vala`, maybe a private method there instead of creating a new utility class?